### PR TITLE
Support setting of SSL version

### PR DESCRIPTION
**Pull request by stanhu**
Closes https://github.com/mikel/mail/issues/659

### CODE REVIEW

**Comment by jeremy** [comment]
Nice! Thanks @stanhu

File: spec/mail/network/delivery_methods/smtp_spec.rb [hunk 1]
```diff
@@ -206,6 +206,28 @@ def redefine_verify_none(new_value)
 
       expect { mail.deliver! }.not_to raise_error
     end
+
+    it "should set verify mode if one is given" do
```
Comment (stanhu) [inline spec/mail/network/delivery_methods/smtp_spec.rb]: Oops, typo here.

**Comment by ye** [comment]
Hi, is there any updates on this PR? Is it ready to be merged?

**Comment by russellsch** [comment]
Would be great if this would get merged! This would sort out some issues I've been having.

**Comment by Brennwert** [comment]
This would also solve my problems. I'd support a merge, please.

**Comment by kyrofa** [comment]
I'd like to see this merged and available as well, please.

**Comment by twk3** [comment]
@jeremy it would be great to get this reviewed/merged. There are more and more servers requiring new tls versions.

**Comment by stanhu** [comment]
It looks like the `ssl_version` is now deprecated. From https://docs.ruby-lang.org/en/2.7.0/OpenSSL/SSL/SSLContext.html:

> Sets the SSL/TLS protocol version for the context. This forces connections to use only the specified protocol version. This is deprecated and only provided for backwards compatibility. Use min_version= and max_version= instead.

**Comment by timkleier** [comment]
@stanhu I noticed your commit to support `min_version` and `max_version`, is there a reason why it was then subsequently reverted? I'd love to see this supported

**Comment by stanhu** [comment]
@timkleier Older Ruby versions use `ssl_version`, while Ruby 2.7+ uses `min_version` and `max_version`. We could probably set this dynamically, but this pull request hasn't received any feedback from the maintainer for years. I'm closing it because of that.

**Pull request closed without merging**